### PR TITLE
SPT: fix setting position of template title

### PR DESCRIPTION
**Pull request by retrofox**
#### Changes proposed in this Pull Request

This PR changes the way of setting the template title scale.

#### Testing instructions

Applies the patch. Check the template title is rightly scaled.

### CODE REVIEW

File: apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/template-selector-preview.js [hunk 1]
```diff
@@ -56,7 +55,26 @@ const TemplateSelectorPreview = ( { blocks, viewportWidth, title } ) => {
...
+				let scale = elStyles.transform.replace( /matrix\((.+)\)$/i, '$1' ).split( ',' );
+				scale = scale && scale.length ? Number( scale[ 0 ] ) : null;
+				scale = isNaN( scale ) ? null : scale;
+
+				if ( titleElement ) {
+					// apply the same transform css rule at template title element.
+					titleElement.style.transform = elStyles.transform;
+				}
+
+				// Try to adjust vertical offset pf the large preview.
```
Comment (obenland) [inline apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/template-selector-preview.js]: `pf` typo

File: apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/template-selector-preview.js [hunk 2]
```diff
@@ -56,7 +55,26 @@ const TemplateSelectorPreview = ( { blocks, viewportWidth, title } ) => {
 			// Try to get the `transform` css rule from the preview container element.
 			const elStyles = window.getComputedStyle( previewContainerEl );
 			if ( elStyles && elStyles.transform ) {
-				setTransform( elStyles.transform ); // apply the same transform css rule to template title.
+				const titleElement = ref.current.querySelector( '.editor-post-title' );
+
+				// pick up scale factor from `transform` css.
```
Comment (obenland) [inline apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/template-selector-preview.js]: Start comment with uppercase?

File: apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/template-selector-preview.js [hunk 3]
```diff
@@ -56,7 +55,26 @@ const TemplateSelectorPreview = ( { blocks, viewportWidth, title } ) => {
...
+				let scale = elStyles.transform.replace( /matrix\((.+)\)$/i, '$1' ).split( ',' );
+				scale = scale && scale.length ? Number( scale[ 0 ] ) : null;
+				scale = isNaN( scale ) ? null : scale;
+
+				if ( titleElement ) {
+					// apply the same transform css rule at template title element.
```
Comment (obenland) [inline apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/template-selector-preview.js]: Start comment with uppercase?

File: apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/template-selector-preview.js [hunk 4]
```diff
@@ -56,7 +55,26 @@ const TemplateSelectorPreview = ( { blocks, viewportWidth, title } ) => {
...
 			if ( elStyles && elStyles.transform ) {
-				setTransform( elStyles.transform ); // apply the same transform css rule to template title.
+				const titleElement = ref.current.querySelector( '.editor-post-title' );
+
+				// pick up scale factor from `transform` css.
+				let scale = elStyles.transform.replace( /matrix\((.+)\)$/i, '$1' ).split( ',' );
+				scale = scale && scale.length ? Number( scale[ 0 ] ) : null;
+				scale = isNaN( scale ) ? null : scale;
+
+				if ( titleElement ) {
```
Comment (obenland) [inline apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/template-selector-preview.js]: Should we group this statement with line 58 above?

**Comment by obenland** [comment]
Functionally this looks much better than what we have currently, nice work @retrofox. Let's get those minor code things fixed up and ship this.

File: apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/template-selector-preview.js (hunk 4 above)
Comment (retrofox) [reply apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/template-selector-preview.js]: Actually, although it's completely related, technically we could apply the factor scale to the title without applying the offset to the large preview block. I've switched some lines of code, but in the end, I prefer to keep the conditions nested.

**Pull request merged**